manifest/manifestlist/manifestlist_test.go

```diff
@@ -109,3 +110,105 @@ func TestManifestList(t *testing.T) {
 		}
 	}
 }
+
+// TODO (mikebrow): add annotations on the index and individual manifests
+var expectedOCIImageIndexSerialization = []byte(`{
```

Comment:
@stevvooe - So how should this PR proceed in that respect? Do you think that is necessary to accept OCI indexes without platforms (the registry doesn't have to accept every valid index or image, right?) If so, what's the right code structure: relax the requirement to include the platform for manifest lists? add conditionals to the code so that the handling of platform is different for manifest lists and OCI image indexes? create a separate type for OCI image indexes? And finally, do you think that needs to be handled prior to landing this PR, or could it be done in a separate PR?